src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -186,19 +231,25 @@ export class WorkspaceSavedObjectsClientWrapper {
       objects: Array<SavedObjectsBulkUpdateObject<T>>,
       options?: SavedObjectsBulkUpdateOptions
     ): Promise<SavedObjectsBulkUpdateResponse<T>> => {
-      const workspaceIds = objects.reduce<string[]>((acc, cur) => {
-        if (this.isRelatedToWorkspace(cur.type)) {
-          acc.push(cur.id);
+      for (const object of objects) {
+        const workspacePermitted = await this.validateWorkspaceAttributesPermitted(
```

Comment:
I'm a bit concern about the bulk action here, it looks the validation of the `objects` list is sequential, which means it validates the object one by one, if there was a long object list, I feel this is going to take unexpected long time. cc @wanglam

Reply:
Maybe we can find some way to do these parallelly, validate 5-10  objects one time. What do you think about that? I think this is out of this PR scope, we can separate one task to optimize it.

Reply:
The logic here is a little tricky. If we want to bulk update the four objects [a, b, c, d], if a and b only have the operation permission on the workspace attribute, and c and d only have the operation permission on the object, then the update operation can still succeed. This means we can only validate based on the granularity of a single object.

Reply:
From the perspective of experience, we can do several verifications on a single object at the same time in a new task , but for the logical integrity, we can only verify based on a single object.